libs/wingsdk/src/std/json.ts

```diff
@@ -97,6 +97,19 @@ export class Json {
     throw new Error("Macro");
   }
 
+  /**
+   * Try to parse a string into a Json
+   *
+   * @macro ((args) => { try { return JSON.parse(args); } catch (err) { return undefined; } })($args$)
+   *
+   * @param str to parse as Json
+   * @returns Json representation of the string or undefined if string is not parsable
+   */
+  public static tryParse(str: string): Json | undefined {
```

Comment:
@Chriscbr - do you think we should return an optional json (`Json?`) here instead of this union type?
Similar to `Bucket.try_get()` (which appears in the [SDK spec](https://docs.winglang.io/reference/wingsdk-spec#bucket), but is not implemented yet).

Reply:
`Json?` as return type is not valid typescript afaik.

Reply:
You are correct, of course. `Json?` Is the desired Wing return type for `try_get`. I'm just wondering what should be the typescript return type that corresponds to that (and it might be that what you've put there is entirely correct - I just wanna see if Chris has an other idea)

Reply:
Thanks for double checking! The way jogold has written this is all good, union with undefined is the best way to represent this in TypeScript I know of 👍

We might want to add some of these type correspondences to our contributing guide, since there are a couple of quirks like this (another example is inflight APIs should always return Promises)

Reply:
Great, thank you both. Resolving my comment